sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -2118,6 +2129,28 @@ std::pair<RT::PiKernel, std::mutex *> ProgramManager::getOrCreateKernel(
                         &(BuildResult->MBuildResultMutex));
 }
 
+bool isDevSupportImgAspects(std::set<RTDeviceBinaryImage *> BinImages,
+                            const device &Dev) {
+  for (RTDeviceBinaryImage *Img : BinImages) {
+    const RTDeviceBinaryImage::PropertyRange &PropRange =
+        Img->getDeviceRequirements();
+    for (RTDeviceBinaryImage::PropertyRange::ConstIterator It : PropRange) {
```

Comment:
Probable the same here. Not sure about nesting level of two lambdas though.

Reply:
I'm not sure if we can do it here, because `It` may be pointing to another property but `aspects`, so we need to sort them before. WDYT?

Reply:
I'll reply near the new code.